azure-resources/Network/expressRouteGateways/recommendations.yaml

```diff
@@ -30,3 +30,24 @@
   learnMoreLink:
     - name: Virtual WAN Monitoring Best Practices
       url: "https://learn.microsoft.com/en-us/azure/virtual-wan/monitoring-best-practices#expressroute-gateway"
+
+- description: Avoid using ExpressRoute circuits for VNet to VNet communication
+  aprlGuid: 560a76a7-8f64-4ce3-ad27-d174468861a1
+  recommendationTypeId: null
+  recommendationControl: High Availability
+  recommendationImpact: Medium
+  recommendationResourceType: Microsoft.Network/expressRouteGateways
+  recommendationMetadataState: Active
+  longDescription: Avoid using ExpressRoute circuits for VNet to VNet communication. Use Virtual WAN to connect VNets in different regions. This feature is enabled at the ExpressRoute circuit level but reflected in the Virtual Wan resource.
+  potentialBenefits: Shortest path between VNETs. Avoids hairpinning traffic through on-premises.
```

Comment:
I wonder if "... through on-premises." should be rewritten as "... through Microsoft edge." or similar.